### PR TITLE
Add ClassInterpolation to typings

### PR DESCRIPTION
**Pull request by Ailrun**
**What**: Add ClassInterpolation to typings

**Why**: Original typings do not accept class interpolations, i.e., `StyledComponent` as interpolation.

**How**: Add ClassInterpolation with private properties.

**Checklist**:
- [N/A] Documentation
- [x] Tests
- [x] Code complete

### CODE REVIEW

File: packages/create-emotion-styled/types/test.tsx [hunk 1]
```diff
@@ -168,13 +168,13 @@ const getEditorLogoColor = (editor: 'vscode' | 'emacs' | 'sublime') => {
...
-}
+};
 
 const StyledCompShorthandWithProps0 = createStyled.div<ShorthandProps>(props => ({
   backgroundColor: getEditorLogoColor(props.editor),
 }));
-const StyledCompShorthandWithProps1 = createStyled.section<ShorthandProps>`
-  backgroundColor: ${props => getEditorLogoColor(props.editor)};
+const StyledCompShorthandWithProps1 = createStyled.section`
+  backgroundColor: ${(props: ShorthandProps) => getEditorLogoColor(props.editor)};
```
Comment (mitchellhamilton) [inline packages/create-emotion-styled/types/test.tsx]: What's the reason for this change?

Comment (Ailrun) [reply packages/create-emotion-styled/types/test.tsx]: @mitchellhamilton It cannot pass check in TS 2.3.
We should add `yarn test:typescript` to TravisCI...

Comment (mitchellhamilton) [reply packages/create-emotion-styled/types/test.tsx]: Yeah, I removed the ts tests from travis because it was crashing and I didn't have time to figure out why, if you want to fix it and add it back [here](https://github.com/emotion-js/emotion/blob/master/package.json#L9) that would be great.

Comment (Ailrun) [reply packages/create-emotion-styled/types/test.tsx]: @mitchellhamilton I will add it after this one will be merged.
